src/main/groovy/org/cadixdev/gradle/licenser/Licenser.groovy

```diff
@@ -53,66 +58,101 @@ class Licenser implements Plugin<Project> {
         this.project = project
 
         project.with {
-            this.extension = extensions.create('license', LicenseExtension, project)
-            extension.header = project.file('LICENSE')
+            this.extension = extensions.create('license', LicenseExtension, project.objects, project)
+            extension.header.set(extension.charset.map { charset -> project.resources.text.fromFile('LICENSE', charset) })
+
+            def headers = objects.listProperty(Header)
+            // TODO: finalizeValueOnRead is not supported on Gradle 5
```

Comment:
Did you mean to do anything differently here?

Reply:
finalizeValueOnRead just provides a bit of protection against incorrect property modification.

though -- I realize now that Groovy is dynamic enough that i should be able to just guard those behind an `if` so that Gradle 6 users get the protection but Gradle 5 still works.